src/main/java/frc/robot/subsystems/leds/LEDStrip.java

```diff
@@ -30,9 +31,24 @@ public void setBuffer() {
      */
     public void addLayer(LEDLayer layer) {
         for (int i = 0; i < ledLength; i++) {
-            if (layer.getLED(i) != null) {
-                ledBuffer.setLED(i, layer.getLED(i));
+            if (layer.getLEDColor(i) != null) {
+                ledBuffer.setLED(i, calcColorWithOpacity(ledBuffer.getLED(i), layer.getLEDColor(i), layer.getLEDOpacity(i)));
             }
         }
     }
+
+    /**
+     * Calculates the desired color when led layering
+     * @param baseColor The color of the lower layer led
+     * @param topColor The color of the led being added
+     * @param opacity The opacity of the top led
+     * @return The color of the led layers combined
+     */
+    public Color calcColorWithOpacity(Color baseColor, Color topColor, double opacity){
+        int r = (int) Math.floor(256 * (((1 - opacity) * baseColor.red) + (opacity * topColor.red)));
+        int g = (int) Math.floor(256 * (((1 - opacity) * baseColor.green) + (opacity * topColor.green)));
+        int b = (int) Math.floor(256 * (((1 - opacity) * baseColor.blue) + (opacity * topColor.blue)));
```

Comment:
Can just use the double constructor that takes in [0.0, 1.0] percentages instead of multiplying by 256 and casting to int (because the int constructor just proceeds to divide by 255 anyways).
```java
return new Color(
    (1 - opacity) * baseColor.red + opacity * topColor.red,
    (1 - opacity) * baseColor.green + opacity * topColor.green,
    (1 - opacity) * baseColor.blue + opacity * topColor.blue
);
```